File: backend/src/models/KpiTemplate.js
const mongoose = require('mongoose');

const kpiTemplateSchema = new mongoose.Schema(
  {
    title: {
      type: String,
      required: [true, 'Title is required'],
      trim: true,
      maxlength: [200, 'Title must not exceed 200 characters'],
    },
    description: {
      type: String,
      required: [true, 'Description is required'],
      trim: true,
      maxlength: [1000, 'Description must not exceed 1000 characters'],
    },
    criteria: {
      type: String,
      required: [true, 'Criteria is required'],
      trim: true,
      maxlength: [500, 'Criteria must not exceed 500 characters'],
    },
    category: {
      type: String,
      trim: true,
      maxlength: [100, 'Category must not exceed 100 characters'],
      default: null,
    },
    isActive: {
      type: Boolean,
      default: true,
    },
    createdBy: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      default: null,
    },
    updatedBy: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      default: null,
    },
  },
  {
    timestamps: true,
  }
);

kpiTemplateSchema.index({ isActive: 1 });
kpiTemplateSchema.index({ category: 1, isActive: 1 });

const KpiTemplate = mongoose.model('KpiTemplate', kpiTemplateSchema);

module.exports = KpiTemplate;

File: frontend/lib/models/kpi_template.dart
class KpiTemplate {
  final String id;
  final String title;
  final String description;
  final String criteria;
  final String? category;
  final bool isActive;
  final DateTime? createdAt;

  KpiTemplate({
    required this.id,
    required this.title,
    required this.description,
    required this.criteria,
    this.category,
    this.isActive = true,
    this.createdAt,
  });

  factory KpiTemplate.fromJson(Map<String, dynamic> json) {
    return KpiTemplate(
      id: (json['_id'] as String?) ?? '',
      title: (json['title'] as String?) ?? '',
      description: (json['description'] as String?) ?? '',
      criteria: (json['criteria'] as String?) ?? '',
      category: json['category'] as String?,
      isActive: (json['isActive'] as bool?) ?? true,
      createdAt: json['createdAt'] != null
          ? DateTime.tryParse(json['createdAt'] as String)
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'description': description,
      'criteria': criteria,
      'category': category,
    };
  }
}
